system/lib/wasmfs/wasmfs.cpp

```diff
@@ -103,6 +104,43 @@ __wasi_errno_t __wasi_fd_close(__wasi_fd_t fd) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+long __syscall_fstat64(long fd, long buf) {
+  // Release FileTable lock after accessing desired open file.
+  auto currentOpenFile = [&fd] {
+    auto fileTable = FileTable::get();
+    return fileTable[fd]->get();
+  }();
+
+  auto file = currentOpenFile.getFile();
+  auto fileHandle = file->get();
```

Comment:
I don't think we actually need to keep the the `std::shared_ptr<File>` around for anything, right?
```suggestion
  auto file = currentOpenFile.getFile()->get();
```

Reply:
Wouldn't we need it for this check `file->is<Directory>()`?